routers/web/auth/oauth.go

```diff
@@ -433,8 +433,21 @@ func AuthorizeOAuth(ctx *context.Context) {
 			log.Error("Unable to save changes to the session: %v", err)
 		}
 	case "":
-		break
+		// "Authorization servers SHOULD reject authorization requests from native apps that don't use PKCE by returning an error message"
+		// https://datatracker.ietf.org/doc/html/rfc8252#section-8.1
+		if !app.ConfidentialClient {
```

Comment:
So this is a break change? Or it's a bug of previous code?

Reply:
Previously Gitea assumed all clients to be confidential. The new column ConfidentialClient defaults to true, so this is backwards compatible.